nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -0,0 +1,127 @@
+            are always compressed to a backup precision which is 8-bit integer by default. All others are quantized
+            whether to NF4 or to a backup precision depending on criteria and the given ratio.
+        :param ratio: the ratio between baseline and backup precisions (e.g. 0.9 means 90% of layers quantized to NF4
+            and the rest to INT8).
+        :param group_size: number of weights (e.g. 128) in the channel dimension
+            that share quantization parameters (scale). The value -1 means no grouping.
+        """
+        super().__init__()
+        self._mode = mode
+        self._group_size = group_size
+        self._ratio = ratio
+        self._backend_entity = None
+        self._algorithm_key = f"CW_{hash(self)}"
+
+    @property
+    def available_backends(self) -> Dict[str, BackendType]:
+        return ALGO_BACKENDS.registry_dict
+
+    def _set_backend_entity(self, model: TModel) -> None:
+        """
+        Creates a helper class with a backed-specific logic of the algorithm.
+
+        :param model: Backend-specific input model.
+        """
+        model_backend = get_backend(model)
+        if model_backend == BackendType.OPENVINO:
+            from nncf.quantization.algorithms.weight_compression.openvino_backend import OVWeightCompressionAlgoBackend
+
+            self._backend_entity = OVWeightCompressionAlgoBackend()
+        else:
+            raise RuntimeError(
+                "Cannot return backend-specific entity because {} is not supported!".format(model_backend.value)
+            )
+
+    def apply(
+        self,
+        model: TModel,
+        graph: NNCFGraph,
+        statistic_points: Optional[StatisticPointsContainer] = None,
+        dataset: Optional[Dataset] = None,
+    ) -> TModel:
+        self._set_backend_entity(model)
+        self._backend_entity.validate_params(self._mode)
+        nodes_to_compress = self._get_nodes_to_compress(graph)
+        transformed_model = self._backend_entity.do_compression(
+            model, nodes_to_compress, self._mode, self._ratio, self._group_size
+        )
+        return transformed_model
+
+    def _get_nodes_to_compress(self, nncf_graph: NNCFGraph) -> List[NNCFNode]:
+        """
+        Collects nodes in the model's graph corresponding to the layers for weight compression.
+
+        :param nncf_graph: NNCFGraph instance.
+        :return: List with the data for each layer.
+        """
+        weighted_metatypes = self._backend_entity.weighted_metatypes
+        ordered_nodes_to_compress = []
+        for node in nncf_graph.topological_sort():
+            is_node_with_weights = self._backend_entity.is_node_with_weights(node)
+            if node.metatype in weighted_metatypes and is_node_with_weights:
+                ordered_nodes_to_compress.append(node)
+        return ordered_nodes_to_compress
+
+    def get_statistic_points(self, model: TModel, graph: NNCFGraph) -> StatisticPointsContainer:
```

Comment:
Should we specify `None` as output value of this function?

Reply:
it'd keep as it's in the interface